darwin/future/core/types/query.py

```diff
@@ -154,17 +162,16 @@ def __init__(
         self.meta_params: dict = meta_params or {}
         self.client = client
         self.filters = filters or []
-        self.results: Optional[List[T]] = None
-        self._changed_since_last: bool = True
+        self.results: dict[int, T] = {}
```

Comment:
This is the core change that allows for Pagination, but all existing functions have been swapped over to preserve the outputs